server/methods/catalog.app-test.js

```diff
@@ -1081,10 +1081,10 @@ describe("core product methods", function () {
       sandbox.stub(Reaction, "hasPermission", () => true);
       let product = addProduct();
       const { isVisible } = product;
-      expect(() => Meteor.call("products/publishProduct", product._id)).to.not.throw(Meteor.Error, /Access Denied/);
+      expect(() => Meteor.call("products/publishProduct", product._id)).to.not.throw();
```

Comment:
I'm not sure if I agree that this is an improvement

Reply:
the problem with passing arguments to `.not.throw` is that it can mask some evils. For example, if you have `.not.throw(Meteor.Error, /Access Denied/)` but the code has a null pointer error, this test passes.

Reply:
Yes, understood, but is that what we are testing? If we want to `expect` that it doesn't throw an error at all maybe that's a different test or at least a separate expectation. Thinking of how to make the test results as useful as possible.

Reply:
If we get a null pointer error, we have some much bigger problems, like using Java

Reply:
haha, I was brought up on C++ and Java... some terms die hard. plus, doesn't NPE have a nicer ring to it than `"Uncaught TypeError: Cannot read property 'method' of undefined"`?

Reply:
ok, if I rewrite the test to be more of a "happy path" test, will that satisfy the concern?
To me, testing that a particular error doesn't get raised is probably fine at the time the test is written, but any future modifications to the code make the test unreliable as a marker for system stability.
As a hypothetical, I could add as the first line of the method `const a = 1/0;` and this test would pass without the code having even reached the line where the error would have been thrown.

Reply:
Yeah, I don't want to be too pedantic about it but I don't want to get into this trap where every test tests everything. Hopefully we have a test that tests whether the function does what it is supposed to, and then a test that permissions work as they supposed to. I am fine with the permission test not failing if the method doesn't work because that's going to be give me a clearer picture of what's happening than if I get 10 tests failing with descriptions like "only allows an admin to execute this function". If I don't have a test that just tests that the function does what it's supposed to then that's a separate problem, but the solution would be to write another test that does more than just test that the method doesn't throw an error.